wallet/src/account/mod.rs

```diff
@@ -321,7 +321,57 @@ impl Account {
         )?;
         // TODO: Randomize inputs and outputs
 
-        let tx = self.sign_transaction(request, db_tx)?;
+        let tx = self.sign_transaction_from_req(request, db_tx)?;
+        Ok(tx)
+    }
+
+    pub fn spend_from_delegation(
+        &mut self,
+        db_tx: &mut impl WalletStorageWriteUnlocked,
+        outputs: Vec<TxOutput>,
+        delegation_id: DelegationId,
+        current_fee_rate: FeeRate,
+    ) -> WalletResult<SignedTransaction> {
+        let delegation_data = self.output_cache.delegation_data(delegation_id)?;
+        let nonce = delegation_data
+            .latest_nonce
+            .increment()
+            .ok_or(WalletError::DelegationNonceOverflow(delegation_id))?;
+
+        let network_fee: Amount = current_fee_rate
+            .compute_fee(
+                tx_size_with_outputs(outputs.as_slice())
+                    + input_signature_size(&delegation_data.destination)?,
+            )
+            .map_err(|_| UtxoSelectorError::AmountArithmeticError)?
+            .into();
+
+        let mut amount_with_fee = network_fee;
+        let mut tx_input = TxInput::Account(AccountOutPoint::new(
+            nonce,
+            Delegation(delegation_id, amount_with_fee),
+        ));
+        let mut input_size = serialization::Encode::encoded_size(&tx_input);
+        loop {
+            amount_with_fee = (network_fee
+                + current_fee_rate
+                    .compute_fee(input_size)
+                    .map_err(|_| UtxoSelectorError::AmountArithmeticError)?
+                    .into())
+            .ok_or(WalletError::OutputAmountOverflow)?;
+            tx_input = TxInput::Account(AccountOutPoint::new(
+                nonce,
+                Delegation(delegation_id, amount_with_fee),
+            ));
+            let new_input_size = serialization::Encode::encoded_size(&tx_input);
+            if new_input_size == input_size {
+                break;
+            }
+            input_size = new_input_size;
+        }
+        let tx = Transaction::new(0, vec![tx_input], outputs)?;
```

Comment:
FYI: there is a potential conflict with PR [#945](https://github.com/mintlayer/mintlayer-core/issues/945). In that PR I restrict paying fees from spending delegation or decommissioning a pool. Meaning tx DelegateStaking -> LockThenTransfer will be rejected because all coins must be locked. In my understanding this should be solved by creating txs like DelegateStaking + Transfer -> LockThenTrnafser + Transfer.

Reply:
never mind the comment, when #945 is properly finished the behaviour won't change for you